api/app/signals/apps/api/v1/filters.py

```diff
@@ -166,6 +180,12 @@ def filter_queryset(self, queryset):
         main_categories = []
         sub_categories = []
 
+        # remove elements if both attribs are not present.
+        if (self.form.cleaned_data.get('area_type_code', None) is None or
+                self.form.cleaned_data.get('area_code', None) is None):
+            self.form.cleaned_data.pop('area_type_code', None)
+            self.form.cleaned_data.pop('area_code', None)
+
```

Comment:
We should add a test for these two fields, see `api/app/tests/apps/api/v1/test_filters.py`.